### ISSUE TITLE
Question about training baselines and 3D input

### ISSUE THREAD

**Issue by geyan21** [intuitive-robots/MoDE_Diffusion_Policy#1]
Hi, Thanks for your great work and clean codebase! I wonder if I can directly train a diffusion policy baseline on the current codebase and data processing pipeline. Or do you have any plans to release that part of the code?

More specifically, I note that `Since MoDE uses action chunking, it needs to load multiple (~10) episode_{}.npz files for each inference. In combination with batching, this results in a large disk bandwidth needed for each iteration (usually ~2000MB/iteration).` If I train a diffusion policy, do I need to follow this step?

In addition, if to train a diffusion policy with point cloud input, where are we supposed to change to support both training and rollout evaluation?

**Comment by mbreuss** [comment]
Hi,

> I can directly train a diffusion policy baseline on the current codebase and data processing pipeline.

Do you mean a DDPM variant? You can directly use the MoDE agent and just substitute the MoDeDiT Transformer part with a CNN-based variant or any other transformer architecture and remove the loss parts of the load balancing. Or what exactly do you want to train? I only have code for my EDM-based variant, as I wanted to compare it with the same diffusion hyperparameters. I would recommend the EDM-based anyway, since in my opinion its better.

>  if to train a diffusion policy with point cloud input

I never trained a 3d variant myself in this codebase. LIBERO does not have a depth-based input as far as I know. For CALVIN you can directly add it here in the config: 
https://github.com/intuitive-robots/MoDE_Diffusion_Policy/blob/main/conf/datamodule/calvin.yaml#L15

You also then need to revise the preprocessing to include the new modality:
https://github.com/intuitive-robots/MoDE_Diffusion_Policy/blob/main/conf/datamodule/transforms/calvin_transforms.yaml

See this link for all depth-based information available for CALVIN: https://github.com/mees/calvin/tree/main/dataset

By changing that you can load the depth maps from CALVIN dataset too. 


> If I train a diffusion policy, do I need to follow this step?

So the reason for that is that the CALVIN dataset stores all modalities for a single timestep in a seperate file. When you train any policy with action chunking this would require your dataloader to open 10 of these timestep files to only extract actions from each file. So by having a seperate file with all actions we can speed up the training. You can adjust the action chunking length how you like it. This is only for faster training. Just run it once and then you dont have to worry anymore about it. 
The framework is flexible to various policies.

Let me know if that helps or if you have any other questions!

**Comment by geyan21** [comment]
Hi, Thanks for your response! I noticed that you have included `Diff-P-CNN` and `Diff-P-T ` baselines in your experiment. I wonder if you use the same action chucking preprocess for these baselines or treat them differently. Since I want to train a `Diff-P-T ` based policy, I'm supposed to directly add an additional `Diff-P-T ` agent right?

**Comment by mbreuss** [comment]
Yes, I don't wanna add them to this repo directly. You can write me an email to moritz.reuss@kit.edu and I can share my implementation for it. Both use the same action chunking logic as MoDe to enable a fair comparison.

**Comment by geyan21** [comment]
Hi, just sent my email. Thanks for your response and happy Christmas!
